### PR TITLE
fix(node:http) Export `validateHeaderName` and `validateHeaderValue` functions

### PR DESCRIPTION
**Pull request by mash-graz**
Modify `_http_outgoing.ts` to support the extended signature of `validateHeaderName()` used since node v19.5.0/v18.14.0 by adding the `label` parameter. (see: https://nodejs.org/api/http.html#httpvalidateheadernamename-label)

Making both validation functions accessible as public exports of `node:http`

Fixes: #22614

### CODE REVIEW

File: ext/node/polyfills/_http_outgoing.ts [hunk 1]
```diff
@@ -820,13 +820,13 @@ Object.defineProperty(OutgoingMessage.prototype, "_headerNames", {
   ),
 });
 
-export const validateHeaderName = hideStackFrames((name) => {
+export const validateHeaderName = hideStackFrames((name:string, label?: string): void => {
```
Comment (aapoalas) [inline ext/node/polyfills/_http_outgoing.ts:823]: **nitpick (blocking):** It seems like the formatting here is not correct, `name:string` is missing a space after the colon. Did you run `./tools/format.js`?

**Review by aapoalas — DISMISSED** [review]
Formatting check fails.

Also, could you add `"test-http-header-validators.js"` string to `tests/node_compat/config.jsonc` line 369. I _think_ that should enable the tests for these.

**Comment by aapoalas** [comment]
Here's a good example of what enabling a node test entails: https://github.com/denoland/deno/pull/22489/files#diff-66a13180ca921c746469f8f0bf6ec0cafab925839d65fdca6379866c4bb7be40

Note this part:
> NOTE: This file should not be manually edited. Please edit `tests/node_compat/config.json` and run `deno task setup` in `tools/node_compat` dir instead.

**Comment by mash-graz** [comment]
The fmt issues should be now fixed and a relevant node_compat test is also activated.

Please review again.

**Comment by mash-graz** [comment]
Please take a look at this more or less trivial modifications and merge them!

They change are so simple, that I didn't open a new branch when I opened the PR, but in the meantime I prepared a lot of other more extensive patches for `deno`, which I con not simply transfer to github, because this PR and it's fixed source branch location are blocking all my further activities... :(

sorry!

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks, good fix.

**Pull request merged**